trailing_spaces.py

```diff
@@ -471,15 +465,12 @@ def freeze_last_version(self, view):
         if file_name and not view.is_scratch() and isfile(file_name):
             encoding = view.encoding()
 
-            if encoding == "Undefined":
-                encoding = view.settings().get("default_encoding", "UTF-8")
-
```

Comment:
I get "Undefined" consistently for some files.
I suspect when the autodetection is not conclusive and the "default_encoding" is used.

Reply:
But then the regexp would fail to match and we would fall back to `view.settings().get("default_encoding", "UTF-8")`

Reply:
This should be done first so that if it does take the "default_encoding" it applies the other checks to it.